src/models/polar.jl

```diff
@@ -0,0 +1,509 @@
+    npq = PS.get(polar.network, PS.NumberOfPQBuses())
+    return 2*npq + npv
+end
+function get(polar::PolarForm, ::NumberOfControl)
+    nref = PS.get(polar.network, PS.NumberOfSlackBuses())
+    npv = PS.get(polar.network, PS.NumberOfPVBuses())
+    return nref + 2*npv
+end
+
+function initial(form::PolarForm{T, VT, AT}, v::AbstractVariable) where {T, VT, AT}
+    pbus = real.(form.network.sbus) |> VT
+    qbus = imag.(form.network.sbus) |> VT
+    vmag = abs.(form.network.vbus) |> VT
+    vang = angle.(form.network.vbus) |> VT
+    return get(form, v, vmag, vang, pbus, qbus)
+end
+
+function get(
+    polar::PolarForm{T, VT, AT},
+    ::State,
+    vmag::VT,
+    vang::VT,
+    pbus::VT,
+    qbus::VT,
+) where {T<:Real, VT<:AbstractVector{T}, AT}
+    npv = PS.get(polar.network, PS.NumberOfPVBuses())
+    npq = PS.get(polar.network, PS.NumberOfPQBuses())
+    nref = PS.get(polar.network, PS.NumberOfSlackBuses())
+    # build vector x
+    dimension = get(polar, NumberOfState())
+    x = VT(undef, dimension)
+    x[1:npq] = vmag[polar.network.pq]
+    x[npq + 1:2*npq] = vang[polar.network.pq]
+    x[2*npq + 1:2*npq + npv] = vang[polar.network.pv]
+
+    return x
+end
+
+function get(
+    polar::PolarForm{T, VT, AT},
+    ::Control,
+    vmag::VT,
+    vang::VT,
+    pbus::VT,
+    qbus::VT,
+) where {T<:Real, VT<:AbstractVector{T}, AT}
+    npv = PS.get(polar.network, PS.NumberOfPVBuses())
+    npq = PS.get(polar.network, PS.NumberOfPQBuses())
+    nref = PS.get(polar.network, PS.NumberOfSlackBuses())
+    pload = polar.active_load
+    # build vector u
+    dimension = get(polar, NumberOfControl())
+    u = VT(undef, dimension)
+    u[1:nref] = vmag[polar.network.ref]
+    # u is equal to active power of generator (Pᵍ)
+    # As P = Pᵍ - Pˡ , we get
+    u[nref + 1:nref + npv] = pbus[polar.network.pv] + pload[polar.network.pv]
+    u[nref + npv + 1:nref + 2*npv] = vmag[polar.network.pv]
+    return u
+end
+
+function get(
+    polar::PolarForm{T, VT, AT},
+    ::Parameters,
+    vmag::VT,
+    vang::VT,
+    pbus::VT,
+    qbus::VT,
+) where {T<:Real, VT<:AbstractVector{T}, AT}
+    npv = PS.get(polar.network, PS.NumberOfPVBuses())
+    npq = PS.get(polar.network, PS.NumberOfPQBuses())
+    nref = PS.get(polar.network, PS.NumberOfSlackBuses())
+    # build vector p
+    dimension = nref + 2*npq
+    p = VT(undef, dimension)
+    p[1:nref] = vang[polar.network.ref]
+    p[nref + 1:nref + npq] = pbus[polar.network.pq]
+    p[nref + npq + 1:nref + 2*npq] = qbus[polar.network.pq]
+    return p
+end
+
+# Bridge with buses' attributes
+function get(polar::PolarForm{T, VT, AT}, ::PS.Buses, ::PS.VoltageMagnitude, x, u, p; V=eltype(x)) where {T, VT, AT}
+    nbus = PS.get(polar.network, PS.NumberOfBuses())
+    npv = PS.get(polar.network, PS.NumberOfPVBuses())
+    npq = PS.get(polar.network, PS.NumberOfPQBuses())
+    nref = PS.get(polar.network, PS.NumberOfSlackBuses())
+    MT = polar.AT
+    vmag = MT{V, 1}(undef, nbus)
```

Comment:
Yeah, this is ugly... Unfortunately that the best I found so far to support both `DualNumber` and `CuVector`. Any suggestion to improve that welcome!